### PR TITLE
Refactor proso_flashcards, the final part

### PR DESCRIPTION
**Pull request by papousek**
What has to be done:
- [ ] fix broken proso_bar
- [ ] migrate services in proso-apps-js

### CODE REVIEW

File: proso_flashcards/models.py [hunk 1]
```diff
@@ -579,21 +284,69 @@ def _create_q(column, value):
...
+    the corresponding item relation.
+    """
+    if instance.id is None:
+        return
+    if len({'term', 'term_id'} & set(instance.changed_fields)) != 0:
+        diff = instance.diff
+        parent = diff['term'][0] if 'term' in diff else diff['term_id'][0]
+        parent_id = parent.item_id if isinstance(parent, Term) else Term.objects.get(pk=parent).item_id
+        child_id = instance.item_id
+        ItemRelation.objects.filter(parent_id=parent_id, child_id=child_id).delete()
```
Comment (thran) [inline proso_flashcards/models.py]: Pustí se signál?

File: proso_flashcards/models.py [hunk 2]
```diff
@@ -579,21 +284,69 @@ def _create_q(column, value):
...
-    environment = get_environment()
-    parent = instance.term.item_id
-    child = instance.item_id
-    environment.delete("child", item=parent, item_secondary=child, symmetric=False)
-    environment.delete("parent", item=child, item_secondary=parent, symmetric=False)
+    """
+    When the given flashcard is deleted, delete also the corresponding item
+    relations.
+    """
+    ItemRelation.objects.filter(child_id=instance.item_id).delete()
```
Comment (thran) [inline proso_flashcards/models.py]: Pustí se signál?

File: proso_models/views.py [hunk 3]
```diff
@@ -98,20 +221,26 @@ def _convert_time_interval(interval):
 
 @allow_lazy_user
 def model(request):
-    if 'items' not in request.GET:
-        return HttpResponseBadRequest('GET parameter "items" has to be specified')
-    item_ids = list(set(map(int, request.GET['items'].split(','))))
+    item_ids = load_query_json(request.GET, 'items', '[]')
```
Comment (thran) [inline proso_models/views.py]: Co to je? chceme to?

File: proso_flashcards/models.py (hunk 1 above)
Comment (papousek) [inline proso_flashcards/models.py]: Ano, https://docs.djangoproject.com/en/1.9/ref/models/querysets/#delete

File: proso_flashcards/models.py (hunk 2 above)
Comment (papousek) [inline proso_flashcards/models.py]: Ano, https://docs.djangoproject.com/en/1.9/ref/models/querysets/#delete

**Pull request closed without merging**